index.js

```diff
@@ -74,6 +75,7 @@ Consumer.prototype.stop = function () {
 Consumer.prototype._poll = function () {
   var receiveParams = {
     QueueUrl: this.queueUrl,
+    MessageAttributeNames: this.messageAttributeNames,
```

Comment:
Just tested this out and it looks like the AWS SDK throws an error if you include this property with a value of `undefined` or anything other than an array. Can we change this so that it's assigned only if `this. messageAttributeNames ` exists? Something like:

```js
  var receiveParams = {
    QueueUrl: this.queueUrl,
    MaxNumberOfMessages: this.batchSize,
    WaitTimeSeconds: 20
  };

  if (this.messageAttributeNames) {
    receiveParams.MessageAttributeNames = this.messageAttributeNames;
  }
```